plugins/aws/iam/trustedCrossAccountRoles.js

```diff
@@ -0,0 +1,93 @@
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Trusted Cross Account Roles',
+    category: 'IAM',
+    description: 'Ensures that only trusted cross-account IAM roles can be used.',
+    more_info: 'IAM roles should be configured to allow access to trusted account IDs.',
+    link: 'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_aws-accounts.html',
+    recommended_action: 'Delete the IAM roles that are associated with untrusted account IDs.',
+    apis: ['IAM:listRoles'],
+    settings: {
+        whitelisted_accounts: {
+            name: 'Allowed cross-account role account IDs',
+            description: 'Return a failing result if cross-account role contains any account ID other than these account IDs',
+            regex: '^[0-9{12}]$',
+            default: [ '123456654321' ]
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var config = {
+            whitelisted_accounts: settings.whitelisted_accounts || this.settings.whitelisted_accounts.default
+        };
+
+        // RegExp to get account ID from 'arn:aws:iam::{account ID}:root' principal
+        var accountRegExp = /(?<=arn:aws:iam::)(.*?)(?=:)/g;
+        var results = [];
+        var source = {};
+        
+        var region = helpers.defaultRegion(settings);
+
+        var listRoles = helpers.addSource(cache, source,
+            ['iam', 'listRoles', region]);
+
+        if (!listRoles) return callback(null, results, source);
+
+        if (listRoles.err || !listRoles.data) {
+            helpers.addResult(results, 3,
+                'Unable to query for IAM roles: ' + helpers.addError(listRoles));
+            return callback(null, results, source);
+        }
+
+        if (!listRoles.data.length) {
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+
+        var awsRolesFound = false;
+
+        listRoles.data.forEach(role => {
+            if (!role.AssumeRolePolicyDocument) {
+                return;
+            }
```

Comment:
you can simplify this 
```suggestion
            if (!role.AssumeRolePolicyDocument) return;
```